zgw_consumers/client.py

```diff
@@ -1,128 +1,99 @@
+from zgw_consumers.constants import AuthTypes
+from zgw_consumers.models import Service
 
-from .settings import get_setting
+# For backwards compatibility:
+from .legacy.client import UnknownService, ZGWClient, get_client_class, load_schema_file
+from .nlx import NLXClient
 
 logger = logging.getLogger(__name__)
 
-IS_OLD_ZDS_CLIENT = zds_client.__version__ < "2.0.0"
-
-Object = Dict[str, Any]
-
-
-def get_client_class() -> type:
-    client_class = get_setting("ZGW_CONSUMERS_CLIENT_CLASS")
-    Client = import_string(client_class)
-    return Client
-
-
-def load_schema_file(file: IO):
-    spec = yaml.safe_load(file)
-    return spec
-
-
-class ZGWClient(Client):
-    def __init__(
-        self,
-        *args,
-        auth_value: Optional[Dict[str, str]] = None,
-        schema_url: str = "",
-        schema_file: IO = None,
-        client_certificate_path=None,
-        client_private_key_path=None,
-        server_certificate_path=None,
-        **kwargs,
-    ):
-        super().__init__(*args, **kwargs)
-
-        self.auth_value = auth_value
-        self.schema_url = schema_url
-        self.schema_file = schema_file
-        self.client_certificate_path = client_certificate_path
-        self.client_private_key_path = client_private_key_path
-        self.server_certificate_path = server_certificate_path
-
-    def fetch_schema(self) -> None:
-        """support custom OAS resolution"""
-        if self.schema_file:
-            logger.info("Loaded schema from file '%s'", self.schema_file)
-            self._schema = load_schema_file(self.schema_file)
-        else:
-            url = self.schema_url or urljoin(self.base_url, "schema/openapi.yaml")
-            logger.info("Fetching schema at '%s'", url)
-            self._schema = schema_fetcher.fetch(url, {"v": "3"})
-
-    def pre_request(
-        self, method: str, url: str, kwargs: Optional[dict] = None, **old_kwargs
-    ):
-        """
-        Add authorization header to requests for APIs without jwt.
-        """
-        kwargs = kwargs or {}
-        if old_kwargs:
-            warnings.warn(
-                "Keyword argument unpacking is removed in zds-client 2.0.",
-                DeprecationWarning,
+
+ClientT = TypeVar("ClientT", bound=NLXClient)
+
+
+def build_client(
+    service: Service, client_factory: type[ClientT] = NLXClient, **kwargs
+) -> ClientT:
+    """
+    Build a client for a given :class:`zgw_consumers.models.Service`.
+    """
+    config_provider = ServiceConfigProvider(service)
+    return client_factory.configure_from(
+        config_provider, nlx_base_url=service.nlx, **kwargs
+    )
+
+
+@dataclass
+class ServiceConfigProvider:
+    """An ``ape-pie`` config provider that will extract session kwargs
+    from a given :class:`zgw_consumers.models.Service`.
+    """
+
+    service: Service
+
+    def get_client_base_url(self) -> str:
+        return self.service.api_root
+
+    def get_client_session_kwargs(self) -> dict[str, Any]:
```

Comment:
this was the method that had the `case` / `match`, right? Can you reinstate that? I'm anticipating future other auth mechanisms, like Oauth2.